PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1679,7 +1679,7 @@ void Notepad_plus::removeDuplicateLines()
 	// whichPart : line head or line tail
 	FindOption env;
 
-	env._str2Search = TEXT("^(.*(\\r?\\n|\\r))(\\1)+");
+	env._str2Search = TEXT("^([^\\r\\n]*(?>\\r?\\n|\\r))(?>\\1)+");

```

Comment:
In some cases this is more economical and allows us to consume more lines when we use `\1` in regex.

Try this line `FF3.46%CRLF` (repet it more times):
3.46%

Without using `?>` it works correct only for 16,3k lines, with `?>` it works correct for 27,3k lines. Still not perfect, but two times better (known issue https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5538). In the future I will try to skip the regex completely  and do it by line iteration.

Reply:
@ArkadiuszMichalski 
> Without using ?> it works correct only for 16,3k lines, with ?> it works correct for 27,3k lines

Are you talking about the memory consuming?
If yes, it should variable, depends on the amount of memory which user has, shouldn't it?

Reply:
@donho
I don't know exactly what this limit depends on. I only checked on one machine. Maybe someone else could check in other machine what the limits are for the same line (just duplicated it many times and use our regex wit/without `?>`  in the Find/Replace window).

This limit also occurs in other commands, where we consume more lines with the pattern `(some_regex_here)+`,  like `removeEmptyLine` or `eol2ws`.

This has already been spotted here https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5538. I don't know if it can be controlled somehow directly in Boost. But if not, then we should probably do it without using regex (because regex operating only on a single/fews line will be very slow).

I'll experiment a bit with the `removeDuplicateLines` function without regex and see how it looks like in practice.

Reply:
@ArkadiuszMichalski 
Thank you for your explanation!